code/modules/client/preferences_savefile.dm

```diff
@@ -374,6 +374,15 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 
 	return TRUE
 
+/datum/preferences/proc/sanitize_be_special(list/input_be_special)
+	var/list/output = list()
+
+	for (var/role in input_be_special)
+		if (role in GLOB.special_roles)
+			output += role
```

Comment:
This is maybe unneeded, but you could get the same effect by doing
`var/list/output = input_be_special & GLOB.special_roles`. Though I don't know if it would maintain order, and it's a bit funky

Reply:
GLOB.special_roles is a dict, this is too funky for me.

Reply:
coward